src/game/server/neo/neo_message.cpp

```diff
@@ -43,12 +53,34 @@ int CNEO_Message::UpdateTransmitState()
     return SetTransmitState(FL_EDICT_ALWAYS);
 }
 
+void CNEO_Message::Think()
+{
+	DisplayTimer();
+	SetNextThink(gpGlobals->curtime + 0.05f);
+}
+
+void CNEO_Message::DisplayTimer()
+{
+	float elapsed = gpGlobals->curtime - m_flTimerStart;
+	char timer[16];
+	V_snprintf(timer, sizeof(timer), "%02d:%02d:%03d", int(elapsed / 60.0f), int(elapsed) % 60, int((elapsed - int(elapsed)) * 1000.0f)); // minutes, seconds, milliseconds
+
+	V_strncpy(m_NetworkedMessageKey.GetForModify(), timer, sizeof(m_NetworkedMessageKey));
```

Comment:
The timer allocation is a bit redundant, since we could format directly to the target buffer:
```cpp
float elapsed = gpGlobals->curtime - m_flTimerStart;
V_snprintf(m_NetworkedMessageKey.GetForModify(), sizeof(m_NetworkedMessageKey),
	"%02d:%02d:%03d", int(elapsed / 60.0f),
	int(elapsed) % 60,
	int((elapsed - int(elapsed)) * 1000.0f)); // minutes, seconds, milliseconds
```